server/src/main/java/org/opensearch/index/shard/RemoteStoreRefreshListener.java

```diff
@@ -82,7 +84,16 @@ public final class RemoteStoreRefreshListener extends ReleasableRetryableRefresh
         REMOTE_REFRESH_RETRY_MAX_INTERVAL_MILLIS
     );
 
-    public static final Set<String> EXCLUDE_FILES = Set.of("write.lock");
+    /**
+     * Checks if a file should be excluded from upload based on exclusion rules.
+     *
+     * @param file the filename to check
+     * @return true if the file should be excluded from upload
+     */
+    public static boolean isFileExcluded(String file) {
+        // Check segments_* pattern as these segment files are not used in Remote Store.
+        return "write.lock".equals(file) || file.startsWith("segments_");
```

Comment:
can we check if there is a constant in Lucene that we can re-use instead of duplication?